backend/routes.js

```diff
@@ -179,12 +179,25 @@ route.GET('/assets/{path*}', {
         return h.continue
       }
     }
+  },
+  files: {
+    relativeTo: path.resolve('dist/assets')
   }
-}, {
-  directory: {
-    path: path.resolve('./dist/assets'),
-    redirectToSlash: true
+}, function (request, h) {
+  const { subpath } = request.params
+  const basename = path.basename(subpath)
+  console.debug(`GET /assets/${subpath}`)
+  // In the build config we told our bundler to use the `[name]-[hash]-cached` template
+  // to name immutable assets. This is useful because `dist/assets/` currently includes
+  // a few files without hash in their name.
+  if (basename.includes('-cached')) {
+    return h.file(subpath, { etagMethod: false })
+      .etag(basename.replace('-cached', ''))
```

Comment:
My OCD & perfectionism is being triggered by this... there's no reason (that I can see) to perform additional operations here on the `basename` vs passing it in unmodified. Will approve though as soon tests pass, and modify it myself in a different PR, unless there are objections

Reply:
Ah, just saw your comment in the other thread (posted after I wrote that above ☝️ )

> Yea that would be simpler, however this also results in more bytes to be sent. Since the full extension if difficult to extract (it would require a regex), I now think that stripping the -chunk part instead is a good compromise.

I don't think sending ~7 bytes more is going to make a difference. To be fair, I also don't think removing the `.replace()` will make much of a difference either. But as they say good coders are lazy - they do the minimum to get the desired results to avoid unnecessary complexity - this is really hard to do in practice (as I just proved by spending ~1 year chasing a more complicated distribution algorithm that in the end we ended up throwing away anyway because in practice it performed worse than the simpler posthumous distribution algorithm).

Reply:
In my other local branch related to this issue, every file in `dist/assets/js` is immutable so the `-cached` infix is no longer needed. But for now you can always remove it if you'd like

Reply:
Yea that's definetely not easy, I can only agree!